src/dashboard/src/components/api/views.py

```diff
@@ -309,103 +290,84 @@ def _completed_units(unit_type='transfer'):
+    )
 
-                # remove trailing slash if not a zipped bag file
-                if not helpers.file_is_an_archive(job.directory):
-                    type_and_directory = type_and_directory[:-1]
+    for job in jobs:
+        # remove standard transfer path from directory (and last character)
+        type_and_directory = job.directory.replace(
+            get_modified_standard_transfer_path() + '/',
+            '',
+            1
+        )
 
-                transfer_watch_directory = type_and_directory.split('/')[0]
-                # Get transfer type from transfer directory
-                transfer_type_directories_reversed = {v: k for k, v in filesystem_ajax_views.TRANSFER_TYPE_DIRECTORIES.items()}
-                transfer_type = transfer_type_directories_reversed[transfer_watch_directory]
+        # remove trailing slash if not a zipped bag file
+        if not helpers.file_is_an_archive(job.directory):
+            type_and_directory = type_and_directory[:-1]
 
-                job_directory = type_and_directory.replace(transfer_watch_directory + '/', '', 1)
+        transfer_watch_directory = type_and_directory.split('/')[0]
+        # Get transfer type from transfer directory
+        transfer_type_directories_reversed = {v: k for k, v in filesystem_ajax_views.TRANSFER_TYPE_DIRECTORIES.items()}
+        transfer_type = transfer_type_directories_reversed[transfer_watch_directory]
 
-                unapproved.append({
-                    'type': transfer_type,
-                    'directory': job_directory,
-                    'uuid': job.sipuuid,
-                })
+        job_directory = type_and_directory.replace(transfer_watch_directory + '/', '', 1)
 
-            # get list of unapproved transfers
-            # return list as JSON
-            response['results'] = unapproved
+        unapproved.append({
+            'type': transfer_type,
+            'directory': job_directory,
+            'uuid': job.sipuuid,
+        })
 
-            if error is not None:
-                response['message'] = error
-                response['error'] = True
-            else:
-                response['message'] = 'Fetched unapproved transfers successfully.'
+    # get list of unapproved transfers
+    # return list as JSON
+    response['results'] = unapproved
 
-                if error is not None:
-                    return helpers.json_response(response, status_code=500)
-                else:
-                    return helpers.json_response(response)
-        else:
-            response['message'] = auth_error
-            response['error'] = True
-            return helpers.json_response(response, status_code=403)
+    if error is not None:
+        response['message'] = error
+        response['error'] = True
     else:
-        return django.http.HttpResponseNotAllowed(permitted_methods=['GET'])
+        response['message'] = 'Fetched unapproved transfers successfully.'
+
+        if error is not None:
```

Comment:
Again, how is it possible for `error` to ever be changed from `None`? Also, why are we checking `error` twice if there is no possibility of it changing after the first check? Lastly, if the `if error is not None` block stays, I think it needs to return the response, right?